File: Array/1920. Build Array from Permutation/Solution.go
func buildArray(nums []int) []int {
	res := make([]int, 0, len(nums))
	for _, val := range nums {
		res = append(res, nums[val])
	}
	return res
}

func buildArray(nums []int) []int {
	res := make([]int, 0, len(nums))
	for i := 0; i < len(nums); i++ {
		res = append(res, nums[nums[i]])
	}
	return res
}

File: Array/1920. Build Array from Permutation/Solution.rs
impl Solution {
    pub fn build_array(nums: Vec<i32>) -> Vec<i32> {
        let mut res=Vec::with_capacity(nums.len());
        for num in nums.iter(){
            res.push(nums[*num as usize]);
        }
        res
    }
}

impl Solution {
    pub fn build_array(nums: Vec<i32>) -> Vec<i32> {
        let mut res=Vec::with_capacity(nums.len());
        for i in 0..nums.len(){
            let t=nums[i] as usize;
            res.push(nums[t]);
        }
        res
    }
}

impl Solution {
    pub fn build_array(nums: Vec<i32>) -> Vec<i32> {
        let mut res=Vec::new();
        for i in 0..nums.len(){
            let t=nums[i] as usize;
            res.push(nums[t]);
        }
        res
    }
}
